Only the first column of slices are zeros

```
slices, xyz_rotated_planes = test_ir.generate_slices(
            map_plane_ones, xy_plane, n_pix, rot_90deg_about_y
        )
        all_ones = np.ones_like(slices[0])
        map_coordinates_artefact = 0
        all_ones[0, :] = map_coordinates_artefact
        all_ones[:, 0] = map_coordinates_artefact
>       assert np.allclose(slices[0], all_ones)
E       assert False
E        +  where False = <function allclose at 0x7f626[60](https://github.com/compSPI/reconstructSPI/runs/5820964148?check_suite_focus=true#step:6:60)90d30>(array([[0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       [0., 1., 1., 1., 1., 1., 1., 1., 1., 1... 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       [0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]]), array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n       [0., 1., 1., 1., 1., 1., 1., 1., 1., 1... 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n       [0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]]))
E        +    where <function allclose at 0x7f[62](https://github.com/compSPI/reconstructSPI/runs/5820964148?check_suite_focus=true#step:6:62)66090d30> = np.allclose
```

--- a/reconstructSPI/iterative_refinement/expectation_maximization.py
+++ b/reconstructSPI/iterative_refinement/expectation_maximization.py
@@ -225,7 +225,7 @@ class IterativeRefinement:
             Shape (n_pix, n_pix, n_pix)
             map normalized by counts.
         """
-        return map_3d * counts / (norm_const + counts**2)
+        return map_3d * counts / (norm_const + counts ** 2)
 
     @staticmethod
     def apply_noise_model(map_3d_f_norm_1, map_3d_f_norm_2):
@@ -342,7 +342,7 @@ class IterativeRefinement:
         axis_pts = np.arange(-n_pix // 2, n_pix // 2)
         grid = np.meshgrid(axis_pts, axis_pts)
 
-        xy_plane = np.zeros((3, n_pix**2))
+        xy_plane = np.zeros((3, n_pix ** 2))
 
         for d in range(2):
             xy_plane[d, :] = grid[d].flatten()
@@ -362,7 +362,7 @@ class IterativeRefinement:
 
         Parameters
         ----------
-        map_3d_f : arr
+        map_3d_f : arr, float (not complex)
             Shape (n_pix, n_pix, n_pix)
         xy_plane : arr
             Array describing xy plane in space.
@@ -382,10 +382,22 @@ class IterativeRefinement:
         xyz_rotated : arr
             Rotated xy planes.
             Shape (n_rotations, 3, n_pix**2)
+
+
+        Notes
+        -----
+        The coordinates are not centered, and the origin/dc component
+        is in map_coordinates. This results in an artefact where the
+        first column of slices[i] is not (always) interpolated,
+        because some rotations, like a 180 deg in xy-plane rotation,
+        do not reach it. It is related to the coordinates going
+        from [n_pix/2,n_pix/2-1] and not [n_pix/2,n_pix/2].
         """
         n_rotations = rots.shape[0]
         slices = np.empty((n_rotations, map_3d_f.shape[0], map_3d_f.shape[1]))
-        xyz_rotated = np.empty((n_rotations, 3, n_pix**2))
+        overwrite_empty_with_zero = 0
+        slices[:, :, 0] = overwrite_empty_with_zero
+        xyz_rotated = np.empty((n_rotations, 3, n_pix ** 2))
         for i in range(n_rotations):
             xyz_rotated[i] = rots[i] @ xy_plane
 
@@ -451,7 +463,7 @@ class IterativeRefinement:
         )
         slices_norm = np.linalg.norm(slices, axis=(1, 2)) ** 2
         particle_norm = np.linalg.norm(particle) ** 2
-        scale = -((2 * sigma**2) ** -1)
+        scale = -((2 * sigma ** 2) ** -1)
         log_bayesian_weights = scale * (slices_norm - 2 * corr_slices_particle)
         offset_safe = log_bayesian_weights.max()
         bayesian_weights = np.exp(log_bayesian_weights - offset_safe)
@@ -569,8 +581,8 @@ class IterativeRefinement:
         a, b, c = center
         nx0, nx1, nx2 = shape
         x0, x1, x2 = np.ogrid[-a : nx0 - a, -b : nx1 - b, -c : nx2 - c]
-        r2 = x0**2 + x1**2 + x2**2
-        mask = r2 <= radius**2
+        r2 = x0 ** 2 + x1 ** 2 + x2 ** 2
+        mask = r2 <= radius ** 2
         if not fill and radius - shell_thickness > 0:
             mask_outer = mask
             mask_inner = r2 <= (radius - shell_thickness) ** 2

--- a/tests/test_expectation_maximization.py
+++ b/tests/test_expectation_maximization.py
@@ -92,7 +92,7 @@ def test_grid_SO3_uniform(test_ir, n_particles):
 def test_generate_xy_plane(test_ir, n_pix):
     """Test generation of xy plane."""
     xy_plane = test_ir.generate_xy_plane(n_pix)
-    assert xy_plane.shape == (3, n_pix ** 2)
+    assert xy_plane.shape == (3, n_pix**2)
 
     n_pix_2 = 2
     plane_2 = np.array([[-1, 0, -1, 0], [-1, -1, 0, 0], [0, 0, 0, 0]])
@@ -122,7 +122,7 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     xy_plane = test_ir.generate_xy_plane(n_pix)
     slices, xyz_rotated_planes = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
     assert slices.shape == (n_particles, n_pix, n_pix)
-    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix ** 2)
+    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix**2)
 
     map_plane_ones = np.zeros((n_pix, n_pix, n_pix))
     map_plane_ones[n_pix // 2] = np.ones((n_pix, n_pix))

--- a/tests/test_expectation_maximization.py
+++ b/tests/test_expectation_maximization.py
@@ -92,7 +92,7 @@ def test_grid_SO3_uniform(test_ir, n_particles):
 def test_generate_xy_plane(test_ir, n_pix):
     """Test generation of xy plane."""
     xy_plane = test_ir.generate_xy_plane(n_pix)
-    assert xy_plane.shape == (3, n_pix**2)
+    assert xy_plane.shape == (3, n_pix ** 2)
 
     n_pix_2 = 2
     plane_2 = np.array([[-1, 0, -1, 0], [-1, -1, 0, 0], [0, 0, 0, 0]])
@@ -120,28 +120,23 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     map_3d = np.ones((n_pix, n_pix, n_pix))
     rots = test_ir.grid_SO3_uniform(n_particles)
     xy_plane = test_ir.generate_xy_plane(n_pix)
-
     slices, xyz_rotated_planes = test_ir.generate_slices(map_3d, xy_plane, n_pix, rots)
-
     assert slices.shape == (n_particles, n_pix, n_pix)
-    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix**2)
+    assert xyz_rotated_planes.shape == (n_particles, 3, n_pix ** 2)
 
     map_plane_ones = np.zeros((n_pix, n_pix, n_pix))
     map_plane_ones[n_pix // 2] = np.ones((n_pix, n_pix))
-
     rot_90deg_about_y = np.array(
         [
             [[0, 0, 1], [0, 1, 0], [-1, 0, 0]],
         ]
     )
-
-    slices, xyz_rotated_planes = test_ir.generate_slices(
-        map_plane_ones, xy_plane, n_pix, rot_90deg_about_y
-    )
     all_ones = np.ones_like(slices[0])
     map_coordinates_artefact = 0
     all_ones[0, :] = map_coordinates_artefact
-    all_ones[:, 0] = map_coordinates_artefact
+    slices, xyz_rotated_planes = test_ir.generate_slices(
+        map_plane_ones, xy_plane, n_pix, rot_90deg_about_y
+    )
     assert np.allclose(slices[0], all_ones)
 
     rot_180deg_about_z = np.array(
@@ -149,13 +144,9 @@ def test_generate_slices(test_ir, n_particles, n_pix):
             [[-1, 0, 0], [0, -1, 0], [0, 0, 1]],
         ]
     )
-
     expected_slice = np.zeros((n_pix, n_pix))
     expected_slice[:, n_pix // 2 - 1] = 1
-
     expected_slice[0, :] = map_coordinates_artefact
-    # expected_slice[:, 0] = map_coordinates_artefact
-
     slices, xyz_rotated_planes = test_ir.generate_slices(
         map_plane_ones, xy_plane, n_pix, rot_180deg_about_z
     )

--- a/tests/test_expectation_maximization.py
+++ b/tests/test_expectation_maximization.py
@@ -106,12 +106,16 @@ def test_generate_xy_plane(test_ir, n_pix):
 def test_generate_slices(test_ir, n_particles, n_pix):
     """Test generation of slices.
 
-    90-degree rotation test.
+    1. shape test.
+
+    2. dc component test. DC component should not change after any rotation.
+
+    3. 90-degree rotation test.
     Map has ones in central xz-plane.
     Rotating by -90 degrees about y
     should produce a slice of ones.
 
-    180-degree rotation test.
+    4. 180-degree rotation test.
     Map has ones in central xz-plane.
     Rotating by 180 degrees about z
     should produce a similar matrix,
@@ -124,6 +128,16 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     assert slices.shape == (n_particles, n_pix, n_pix)
     assert xyz_rotated_planes.shape == (n_particles, 3, n_pix ** 2)
 
+    map_3d_dc = np.zeros((n_pix, n_pix, n_pix))
+    rand_val = np.random.random_uniform(low=1, high=2)
+    map_3d_dc[n_pix // 2, n_pix // 2, n_pix // 2] = rand_val
+    slices, xyz_rotated_planes = test_ir.generate_slices(
+        map_3d_dc, xy_plane, n_pix, rots
+    )
+    assert np.allclose(
+        slices[:, n_pix // 2, n_pix // 2], rand_val * np.ones(len(slices))
+    )
+
     map_plane_ones = np.zeros((n_pix, n_pix, n_pix))
     map_plane_ones[n_pix // 2] = np.ones((n_pix, n_pix))
     rot_90deg_about_y = np.array(
@@ -133,7 +147,7 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     )
     all_ones = np.ones_like(slices[0])
     map_coordinates_artefact = 0
-    all_ones[0, :] = map_coordinates_artefact
+    all_ones[:, 0] = map_coordinates_artefact
     slices, xyz_rotated_planes = test_ir.generate_slices(
         map_plane_ones, xy_plane, n_pix, rot_90deg_about_y
     )
@@ -146,18 +160,11 @@ def test_generate_slices(test_ir, n_particles, n_pix):
     )
     expected_slice = np.zeros((n_pix, n_pix))
     expected_slice[:, n_pix // 2 - 1] = 1
-    expected_slice[0, :] = map_coordinates_artefact
+    expected_slice[:, 0] = map_coordinates_artefact
     slices, xyz_rotated_planes = test_ir.generate_slices(
         map_plane_ones, xy_plane, n_pix, rot_180deg_about_z
     )
     assert np.allclose(slices[0], expected_slice)
-
-    map_3d_dc = np.zeros((n_pix, n_pix, n_pix))
-    map_3d_dc[n_pix // 2, n_pix // 2, n_pix // 2] = 1
-    slices, xyz_rotated_planes = test_ir.generate_slices(
-        map_plane_ones, xy_plane, n_pix, rot_90deg_about_y
-    )
-    assert np.allclose(slices[:, n_pix // 2, n_pix // 2], np.ones(len(slices)))
 
 
 def test_apply_ctf_to_slice(test_ir, n_pix):

